src/frontend/web_application/src/scenes/Contact/components/PublicKeyForm/presenter.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Field } from 'redux-form';
+import { withI18n, Trans } from '@lingui/react';
+import renderReduxField from '../../../../services/renderReduxField';
+import { Button, Confirm, FieldErrors, Fieldset, FormColumn, FormGrid, FormRow, Icon, Legend, TextFieldGroup as TextFieldGroupBase, TextareaFieldGroup as TextareaFieldGroupBase } from '../../../../components';
+
+import './style.scss';
+
+const TextFieldGroup = renderReduxField(TextFieldGroupBase);
+const TextareaFieldGroup = renderReduxField(TextareaFieldGroupBase);
+
+@withI18n()
+class PublicKeyForm extends PureComponent {
+  static propTypes = {
+    errors: PropTypes.arrayOf(PropTypes.string),
+    i18n: PropTypes.shape({}).isRequired,
+    contactId: PropTypes.string.isRequired,
+    publicKey: PropTypes.shape({}),
+    handleSubmit: PropTypes.func.isRequired,
+    form: PropTypes.string.isRequired,
+    createPublicKey: PropTypes.func.isRequired,
+    updatePublicKey: PropTypes.func.isRequired,
+    deletePublicKey: PropTypes.func.isRequired,
+    onSuccess: PropTypes.func,
```

Comment:
it is not an agnostic lib so we can required it because we are sure the parent needs this
(but I might be wrong if your some cases when it is not required)

Reply:
This is kind of a reflex : a component might be reused elsewhere in the project, in a slightly different context. I really don't like to tie a component to its parent if I don't need to. At its root, the very concept of component is re-usability.

Reply:
I understand what you mean, I have this reflex too; but like this case it is almost 90% sure this kind of components will not be reused and it will be more efficient to be straight forward and keep it very specific.

Agree too to say it is not big deal here.
I only want to prevent hours of debugging because for example someone removed or made a typo in the handle on `onsuccess` and get stucked on "Hey, why this doesn't work no more?!" without any warning.